app/code/Magento/Inventory/Model/ReservationBuilder.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model;
+
+use Magento\InventoryApi\Api\Data\ReservationInterface;
+
+/**
+ * Used to instantiate ReservationInterface objects
+ */
+class ReservationBuilder implements ReservationBuilderInterface
+{
+    /**
+     * @var int
+     */
+    private $reservationId;
+
+    /**
+     * @var int
+     */
+    private $stockId;
+
+    /**
+     * @var string
+     */
+    private $sku;
+
+    /**
+     * @var float
+     */
+    private $quantity;
+
+    /**
+     * @var string
+     */
+    private $metadata;
+
+    /**
+     * Object Manager instance
+     *
+     * @var \Magento\Framework\ObjectManagerInterface
+     */
+    protected $_objectManager = null;
+
+    /**
+     * ReservationBuilder constructor.
+     *
+     * @param \Magento\Framework\ObjectManagerInterface $objectManager
+     */
+    public function __construct(
+        \Magento\Framework\ObjectManagerInterface $objectManager
+    ) {
+        $this->_objectManager = $objectManager;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setReservationId($reservationId): ReservationBuilder
+    {
+        $this->reservationId = $reservationId;
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setStockId(int $stockId): ReservationBuilder
+    {
+        $this->stockId = $stockId;
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setSku(string $sku): ReservationBuilder
+    {
+        $this->sku = $sku;
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setQuantity(float $quantity): ReservationBuilder
+    {
+        $this->quantity = $quantity;
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setMetadata($metadata): ReservationBuilder
+    {
+        $this->metadata = $metadata;
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function build(): ReservationInterface
+    {
+        $data = [
+            ReservationInterface::RESERVATION_ID => $this->reservationId,
+            ReservationInterface::STOCK_ID => $this->stockId,
+            ReservationInterface::SKU => $this->sku,
+            ReservationInterface::QUANTITY => $this->quantity,
+            ReservationInterface::METADATA => $this->metadata,
+        ];
+
+        $arguments = [
+            'data' => $data,
+        ];
+
+        return $this->_objectManager->create(ReservationInterface::class, $arguments);
```

Comment:
after the object creation builder should clean its own state,
because we will use this builder to create several entities in a row (one by one).
So, after the first Reservation is created the state of builder should not be polluted, which will affect creation of second entity

Reply:
I can add a private `reset()` method that cleans the builder state; I can call it after the creation of a Reservation instance before returning it. Does it sound good?

Reply:
@aleron75  yes, it sounds reasonable